types/index.d.ts

```diff
@@ -0,0 +1,137 @@
+export type Leagues = "little" | "great" | "ultra" | "master";
+
+export interface Stats {
+    attack: number;
+    defense: number;
+    stamina: number;
+}
+
+export interface PvPStat {
+    rank: number;
+    attack: number;
+    defense: number;
+    stamina: number;
+    cap: number;
+    value: number;
+    level: number;
+    cp: number;
+    percentage?: number;
+    capped?: boolean;
+    evolution?: number;
+    form?: number;
+}
+
+export interface RankEntry {
+    pokemon: number;
+    form?: number;
+    evolution?: number;
+    cap: number;
+    capped?: boolean;
+    level: number;
+    cp: number;
+    value: number;
+    percentage: number;
+}
+
+export interface RankResult {
+    value: number;
+    level: number;
+    cp: number;
+    percentage: number;
+    rank: number;
+}
+
+export interface LeagueOptions {
+    cap: number | null;
+    little?: boolean;
+}
+
+export interface CachingStrategy {
+    (): [LRUCache, boolean];
+}
+
+export interface LRUCache {
+    get(key: string): object;
+    set(key: string, value: object): void;
+}
+
+export interface OhbemOptions {
+    leagues?: Record<string, LeagueOptions | number | null>;
+    levelCaps?: number[];
+    pokemonData?: PokemonData;
```

Comment:
I think `PokemonData` already has a ts? @TurtIeSocks

Reply:
Yes yours is better. Thanks again! I will take a closer look and merge when I get a chance.